examples/src/main/scala/org/apache/spark/examples/ml/KMeansExample.scala

```diff
@@ -50,6 +51,7 @@ object KMeansExample {
     val sc = new SparkContext(conf)
     val sqlContext = new SQLContext(sc)
 
+    // $example on$
     // Loads data
     val rowRDD = sc.textFile(input).filter(_.nonEmpty)
       .map(_.split(" ").map(_.toDouble)).map(Vectors.dense).map(Row(_))
```

Comment:
As long as this file is being changed, let's simplify the example a little:
Rather than mapping to Row and creating a schema explicitly, how about following what is done here: [https://github.com/apache/spark/blob/c119a34d1e9e599e302acfda92e5de681086a19f/examples/src/main/scala/org/apache/spark/examples/ml/BinarizerExample.scala].  The "label" could be replaced by an "id"